src/controllers/controller.bar.js

```diff
@@ -154,7 +160,7 @@ module.exports = function(Chart) {
 			var vscale = me.getValueScale();
 			var base = vscale.getBasePixel();
 			var horizontal = vscale.isHorizontal();
-			var ruler = me._ruler || me.getRuler();
+			var ruler = (!me._ruler || index >= me._ruler.length) ? me.getRuler() : me._ruler;
```

Comment:
Ok, I will leave the ruler to cache `{start, end, pixels, stackCount, scale}` and let calculateBarIndexPixels() calculate the bar size using those values.